lib/rubocop/cop/lint/shadowed_exception.rb

```diff
@@ -72,17 +68,17 @@ def on_rescue(node)
         private
 
         def offense_range(rescues)
-          first_rescue = rescues.first
-          last_rescue = rescues.last
-          last_exceptions, = *last_rescue
-          # last_rescue clause may not specify exception class
-          end_pos = if last_exceptions
-                      last_exceptions.loc.expression.end_pos
-                    else
-                      last_rescue.loc.keyword.end_pos
-                    end
-
-          range_between(first_rescue.loc.expression.begin_pos, end_pos)
+          shadowing_rescue = find_shadowing_rescue(rescues)
+          expression = shadowing_rescue.loc.expression
+          range_between(expression.begin_pos, expression.end_pos)
+        end
+
+        def rescued_groups_for(rescues)
+          rescues.each_with_object([]) do |group, exceptions|
```

Comment:
Since the code is changing anyways, I think this can use `map` instead of `each_with_object`.

```ruby
rescues.map do |group|
  rescue_group, = *group
  evaluate_exceptions(rescue_group)
end
```